packages/core-data/src/actions.js

```diff
@@ -115,14 +115,100 @@ export function receiveEmbedPreview( url, preview ) {
 	};
 }
 
+/**
+ * Returns an action object that triggers an
+ * edit to an entity record.
+ *
+ * @param {string} kind           Kind of the edited entity record.
+ * @param {string} name           Name of the edited entity record.
+ * @param {number} recordId       Record ID of the edited entity record.
+ * @param {Object} edits          The edits.
+ *
+ * @return {Object} Action object.
+ */
+export function* editEntityRecord( kind, name, recordId, edits ) {
+	const { transientEdits = {}, mergedEdits = {} } = yield select( 'getEntity', kind, name );
+	const record = yield select( 'getEntityRecord', kind, name, recordId );
+	const editedRecord = yield select(
+		'getEditedEntityRecord',
+		kind,
+		name,
+		recordId
+	);
+
+	const edit = {
+		kind,
+		name,
+		recordId,
+		// Clear edits when they are equal to their persisted counterparts
+		// so that the property is not considered dirty.
+		edits: Object.keys( edits ).reduce( ( acc, key ) => {
+			const value = mergedEdits[ key ] ?
+				merge( record[ key ], edits[ key ] ) :
```

Comment:
This works a bit differently than what we currently have in the editor package. I think we only store in the edits reducer the modified sub properties and do the merge in the selectors. This might be simpler though.

Reply:
Yeah, we need this for undos to work in this context.